web/src/beta/lib/reearth-ui/components/NumberInput/index.tsx

```diff
@@ -0,0 +1,132 @@
+import { useState, useCallback, useEffect, FC } from "react";
+
+import { fonts, styled } from "@reearth/services/theme";
+
+export type NumberInputProps = {
+  value?: number;
+  size?: "normal" | "small";
+  disabled?: boolean;
+  placeholder?: string;
+  appearance?: "readonly";
+  unit?: string;
+  onChange?: (value?: number) => void;
+  onBlur?: (value?: number) => void;
+};
+
+export const NumberInput: FC<NumberInputProps> = ({
+  value,
+  size = "normal",
+  disabled,
+  placeholder,
+  appearance,
+  unit,
+  onChange,
+  onBlur,
+}) => {
+  const [currentValue, setCurrentValue] = useState<number | undefined>(value);
+  const [isFocused, setIsFocused] = useState(false);
+
+  useEffect(() => {
+    setCurrentValue(value);
+  }, [value]);
+
+  const handleChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      const newValue = e.currentTarget.value;
+      const parsedValue = parseFloat(newValue);
+      setCurrentValue(parsedValue);
+      onChange?.(parsedValue);
+    },
+    [onChange],
+  );
+
+  const handleBlur = useCallback(
+    (e: React.SyntheticEvent<HTMLInputElement>) => {
+      const newValue = parseFloat(e.currentTarget.value) ?? value;
+      setIsFocused(false);
+      onBlur?.(newValue);
+    },
+    [value, onBlur],
+  );
+
+  const handleFocus = useCallback(() => {
+    setIsFocused(true);
+  }, []);
+
+  // This function prevent defaut browser clear input on these keys pressed
+  // https://stackoverflow.com/questions/49869624/numeric-value-gets-clear-in-we-press-letter-e-in-input-type-number
+  const handleKeyDown = useCallback((e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "e" || e.key === "+" || e.key === "-") {
+      e.preventDefault();
+    }
+  }, []);
```

Comment:
This prevent user input negative number value as well. Maybe we need to think more for the solution.